mavros/src/plugins/gps_rtk.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * @brief GPS RTK plugin
+ * @file gps_rtk.cpp
+ * @author Alexis Paques <alexis.paques@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2018 Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros_msgs/RTCM.h>
+#include <algorithm>
+
+namespace mavros {
+namespace std_plugins {
+/**
+ * @brief GPS RTK plugin
+ *
+ * Publish the RTCM messages from ROS to the Pixhawk
+ */
+class GpsRtkPlugin : public plugin::PluginBase {
+public:
+	GpsRtkPlugin() : PluginBase(),
+		gps_rtk_nh("~gps_rtk")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+		gps_rtk_sub = gps_rtk_nh.subscribe("send_rtcm", 10, &GpsRtkPlugin::rtk_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return {};
+	}
+
+private:
+	ros::NodeHandle gps_rtk_nh;
+	ros::Subscriber gps_rtk_sub;
+
+	void rtk_cb(const mavros_msgs::RTCM::ConstPtr &msg)
+	{
+		const int maxMessageLength = 180;	// Should be replaced by MAVLINK_MSG_GPS_RTCM_DATA_FIELD_DATA_LEN
```

Comment:
I think you can use `auto max_len = mavlink::common::msg::GPS_RTCM_DATA::data::size()` as it constexpr.

Reply:
Also i don't like CamelCase for anything except class names.

Reply:
I tried to use `mavlink::common::msg::GPS_RTCM_DATA::data::size()` but it fails to compile: 

```bash
/home/alexis/Workspace/src/mavros/mavros/src/plugins/gps_rtk.cpp: In member function ‘void mavros::std_plugins::GpsRtkPlugin::rtk_cb(const ConstPtr&)’:
/home/alexis/Workspace/src/mavros/mavros/src/plugins/gps_rtk.cpp:52:53: error: ‘mavlink::common::msg::GPS_RTCM_DATA::data’ is not a class, namespace, or enumeration
   auto max_len = mavlink::common::msg::GPS_RTCM_DATA::data::size();
```

As data is a vector, I tried: `auto max_len = mavlink::common::msg::GPS_RTCM_DATA::data.size();` but it fails as it is not instanciated.

```bash
/home/alexis/Workspace/src/mavros/mavros/src/plugins/gps_rtk.cpp: In member function ‘void mavros::std_plugins::GpsRtkPlugin::rtk_cb(const ConstPtr&)’:
/home/alexis/Workspace/src/mavros/mavros/src/plugins/gps_rtk.cpp:52:55: error: invalid use of non-static data member ‘mavlink::common::msg::GPS_RTCM_DATA::data’
   auto max_len = mavlink::common::msg::GPS_RTCM_DATA::data.size();
                                                       ^
In file included from /opt/ros/kinetic/include/mavlink/v2.0/common/common.hpp:1310:0,
                 from /home/alexis/Workspace/src/mavros/libmavconn/include/mavconn/mavlink_dialect.h:26,
                 from /home/alexis/Workspace/src/mavros/libmavconn/include/mavconn/interface.h:37,
                 from /home/alexis/Workspace/src/mavros/mavros/include/mavros/mavros_plugin.h:24,
                 from /home/alexis/Workspace/src/mavros/mavros/src/plugins/gps_rtk.cpp:17:
/opt/ros/kinetic/include/mavlink/v2.0/common/./mavlink_msg_gps_rtcm_data.hpp:24:30: note: declared here
     std::array<uint8_t, 180> data; /*< RTCM message (may be fragmented) */
```

Reply:
It works as: 

```cpp
	mavlink::common::msg::GPS_RTCM_DATA rtcm_data;
	const int max_len = rtcm_data.data.size(); // 180 bytes
```

Yet, this would be useful if we are sending the message once; Here we could send it 4 times. 
I guess that  `UAS_FCU(m_uas)->send_message(rtcm_data);` is non blocking, so there could be a race condition issue if I use the same object. 

Is it safe to keep the same object and send it 4 times with different data?